File: Search/SqrtX/SqrtX.cc
class Solution {
    public:
        int mySqrt(int x) {
            if(x < 0)
                return -1;
            else if (x == 0)
                return 0;


            int start = 0;
            int end = x;
            long long mid = 0;
            while(start + 1 < end) {
                mid = start + (end - start) / 2;
                if(mid * mid == x)
                    return mid;
                else if (mid * mid < x)
                    start = mid;
                else
                    end = mid; 
            }
            return start;
        }
};
        

File: Search/SqrtX/SqrtX.py
class Solution:
    def mySqrt(self, x):
        if x < 0:
            return -1
        
        if x == 0:
            return 0

        start, end = 1, x
        while start + 1 < end:
            mid = (start + end) / 2
            if mid ** 2 == x:
                return mid
            elif mid ** 2 < x:
                start = mid
            else:
                end = mid

        return start

        
